packages/ui-components/src/components/ReferenceTooltip.tsx

```diff
@@ -46,39 +46,64 @@ const StyledToolTip = withStyles(theme => ({
   },
 }))(Tooltip);
 
-export interface ReferenceProps {
-  text?: string;
-  name?: string;
-  link?: string;
+interface PlaintextReferenceProps {
+  text: string;
+  name?: never;
+  link?: never;
+}
+interface LinkReferenceProps {
+  text?: never;
+  name: string;
+  link: string;
 }
 
-const Content = ({ text = '', name = '', link = '' }: ReferenceProps) => {
-  if (text) {
-    return (
-      <TextCaption variant="caption">
-        <span>{text} </span>
-      </TextCaption>
-    );
+/**
+ * Props for the reference prop of the ReferenceTooltip component. It can have either a piece of
+ * plaintext to display in the tooltip, or a named link; but not both.
+ */
+export type ReferenceProps = PlaintextReferenceProps | LinkReferenceProps;
+
+const isPlaintextReferenceProp = (obj: ReferenceProps): obj is PlaintextReferenceProps =>
+  'text' in obj && !('name' in obj) && !('link' in obj);
```

Comment:
Hm, I also need these in types in `packages/types/src/types/requests/tupaia-web-server/MapOverlaysRequest.ts` (specifically `PlaintextReferenceProps`). Is this still the most appropriate place to define this type and type guard?

Reply:
Had a think about this, and because the reference tooltip is used by a few other things (not just MapOverlaysRequest), there isn’t any single natural place to define it in the orchestration server. Here feels the least bad to me